src/Address.svelte

```diff
@@ -1,51 +1,49 @@
 <script lang="typescript">
   import { onMount } from 'svelte';
+  import { navigate } from 'svelte-routing';
   import { link } from 'svelte-routing';
   import { ethers } from 'ethers';
-  import { explorerLink, safeLink } from '@app/utils';
-  import Blockies from '@app/Blockies.svelte';
+  import { safeLink, identifyAddress, formatAddress, AddressType } from '@app/utils';
   import Loading from '@app/Loading.svelte';
+  import Avatar from "@app/Avatar.svelte";
+  import Error from '@app/Error.svelte';
   import type { Config } from '@app/config';
-  import { identifyAddress, formatAddress, AddressType } from '@app/utils';
+  import type { Registration } from '@app/base/registrations/registrar';
+  import { getRegistration } from '@app/base/registrations/registrar';
 
   export let address: string;
   export let config: Config;
   export let resolve = false;
   export let noBadge = false;
+  export let noAvatar = false;
   export let compact = false;
 
   let checksumAddress = compact
     ? formatAddress(address)
     : ethers.utils.getAddress(address);
   let addressType: AddressType | null = null;
   let addressName: string | null = null;
+  let getInfo: Promise<Registration | null>;
 
-  onMount(() => {
-    identifyAddress(address, config).then(typ => addressType = typ);
+  onMount(async () => {
+    identifyAddress(address, config).then((t: AddressType) => addressType = t);
     if (resolve) {
-      config.provider.lookupAddress(address).then(name => addressName = name);
+      addressName = await config.provider.lookupAddress(address);
+      getInfo = getRegistration(addressName, config);
```

Comment:
Note that `getInfo` is only set here if `resolve == true`, but in the HTML below, we don't render anything unless getInfo resolves!

Reply:
I think it's preferrable to leave that logic the way it currently is, where the address rendering doesn't depend on anything async from resolving.

Reply:
I was looking for someway to show the page only if finished loading so we don't see rapid changes of information.. Will see if I come up with a better way

Reply:
Yeah I guess if it falls back to the address in case of error, that would be fine. Otherwise I'd say don't worry about it for now.

Reply:
I removed the waiting on a result, there is some flickering while the information loads, but it's acceptable.